src/3p-frame.js

```diff
@@ -212,22 +212,30 @@ export function resetBootstrapBaseUrlForTesting(win) {
  * @return {string}
  */
 export function getDefaultBootstrapBaseUrl(parentWindow, opt_srcFileBasename) {
-  if (parentWindow.defaultBootstrapBaseUrl) {
-    return parentWindow.defaultBootstrapBaseUrl;
-  }
   const srcFileBasename = opt_srcFileBasename || 'frame';
+  // Cached at filename level to ensure different url is returned.
+  parentWindow.defaultBootstrapBaseUrl =
+      parentWindow.defaultBootstrapBaseUrl || {};
+  if (parentWindow.defaultBootstrapBaseUrl[srcFileBasename]) {
+    return parentWindow.defaultBootstrapBaseUrl[srcFileBasename];
+  }
   if (getMode().localDev || getMode().test) {
     if (overrideBootstrapBaseUrl) {
-      return parentWindow.defaultBootstrapBaseUrl = overrideBootstrapBaseUrl;
+      return parentWindow.defaultBootstrapBaseUrl[srcFileBasename] =
+          overrideBootstrapBaseUrl;
     }
-    return parentWindow.defaultBootstrapBaseUrl = getAdsLocalhost(parentWindow)
-        + '/dist.3p/'
-        + (getMode().minified ? `$internalRuntimeVersion$/${srcFileBasename}`
-            : `current/${srcFileBasename}.max`)
-        + '.html';
+    return parentWindow.defaultBootstrapBaseUrl[srcFileBasename] =
+        getAdsLocalhost(parentWindow)
+          + '/dist.3p/'
+          + (getMode().minified ? `$internalRuntimeVersion$/${srcFileBasename}`
+              : `current/${srcFileBasename}.max`)
+          + '.html';
   }
-  return parentWindow.defaultBootstrapBaseUrl = 'https://' +
-      getSubDomain(parentWindow) +
+  // Ensure same sub-domain is used despite potentially different file.
+  parentWindow.defaultBootstrapSubDomain =
```

Comment:
might be better to put this caching logic directly within getSubDomain, and then update the doc on that function.

Reply:
I debated that but the function's doc indicates it expects to only ever be called once and I'm assuming that solution was considered previously and not used.

Reply:
right but now that doc is incorrect, it's going to be called multiple times. It seems like the logic that handles that should be in getSubDomain, no? I don't get why it would be better to keep that here.

Reply:
whoops ignore that first point of the last comment. discussed offline.